Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Image/StreamingImageAsset.h

```diff
@@ -143,6 +147,8 @@ namespace AZ
             //! mip chain index, not the level. And will fail for any level
             //! that resides in the tail mip chain.
             const ImageMipChainAsset* GetImageMipChainAsset(AZ::u32 mipLevel) const;
+
+            AZStd::unordered_set<AZ::Name> m_tags;
```

Comment:
Suggestion - Looking at the places this is used I'm not sure it needs to be an unordered set. Vector is faster to iterate through and likely to be faster for lookups with small numbers of tags (which I assume is the case here). Only thing I think you'd need to be careful about is duplicates being added in `AddTag()` but that's a simple check.

Reply:
Yes I went for a set just to quickly ensure there won't be any duplicate. Vector is faster to iterate but I doubt it'll change anything since tags are iterated only once on image creation